### PR TITLE
[XConfessions] Support sceneByName and sceneByQueryFragment and fix scene scraper by url

### PR DESCRIPTION
**Pull request by TgSeed**
Fixes:
- Scene Details
- Scene Performers
- Scene Tags
- Scene Image

### CODE REVIEW

**Comment by TgSeed** [comment]
[afa0cfc](https://github.com/stashapp/CommunityScrapers/pull/1140/commits/afa0cfcac379a9931ab90840eb9f6532a54c0422) Added support for `sceneByName` and `sceneByQueryFragment` by using Python script.
Kept the scene data scraper in `.yml` file to keep the python script almost dependency free (except [requests](https://pypi.org/project/requests/) i was needing to use [lxml](https://pypi.org/project/lxml/)) and easier for others to edit/fix in future

**Comment by TgSeed** [comment]
[337d901](https://github.com/stashapp/CommunityScrapers/pull/1140/commits/337d901b71f0344c0a554dfc02c9f14ff955f9c6) commit makes stash to set URL for newly created performers by the scraped scene.

[013438b](https://github.com/stashapp/CommunityScrapers/pull/1140/commits/013438b240dd01d10a4880d6a52bf3d5e3f47d67) makes it possible to scrape performer by the URL that is already set.

**Comment by TgSeed** [comment]
[c875548](https://github.com/stashapp/CommunityScrapers/pull/1140/commits/c875548660ba338bffacc643e8865aab023ab014) adds ability to search for performer by name
NOTE: while the search results include gender (the JSON response), the web page doesn't show or include the gender in anywhere! as far as there is no way to pass the gender from `performerByName script` to `performerByURL scrapeXPath` or the web page doesn't show gender, scraper cannot set it for the performer.

File: scrapers/XConfessions.yml [hunk 1]
```diff
@@ -62,4 +89,55 @@ xPathScrapers:
...
+          - replace:
+              - regex: ^(.*?),\s
+                with: ""
+      Details: $data//div[contains(@class, "description-block")]/p
+      Images:
+        selector: $data//img[contains(@class, "image") and contains(@class, "absolute") and contains(@class, "w-full")]/@srcset
+        postProcess:
+          - replace:
+              - regex: ^([^?]+)\?.*
+                with: $1
```
Comment (bnkai) [inline scrapers/XConfessions.yml:116]: copy paste error ? the image is already present below

Comment (TgSeed) [reply scrapers/XConfessions.yml]: Oof
That was just for my fork/PR. forget to remove for the PR
Removed.

**Review by JackDawson94 — COMMENTED** [review]
Tested
- Performer by Name & Url work as intended
- Scene search & scrape works as intended

Scraping Movies generates some errors when I feed urls like https://xconfessions.com/film/my-ass
Unless there is another type of urls for movies ? In that case the scraper should filter for that type of url.


Not sure of the value of having a Movie scraper, I haven't watched much of their work but I believe xconfessions only provides single-scene movies right ?

**Comment by Maista6969** [comment]
Closing this as superseded by #1758

**Pull request closed without merging**